src/cursor/find_cursor.ts

```diff
@@ -49,6 +49,16 @@ export class FindCursor extends AbstractCursor {
     }
   }
 
+  /**
+   * Returns a new uninitialized copy of this cursor, with options matching those that have been set on the current instance
+   */
+  clone(): FindCursor {
+    return new FindCursor(this.topology, this.namespace, this[kFilter], {
+      ...this[kBuiltOptions],
+      ...this.cursorOptions
+    });
+  }
```

Comment:
I think you can still make this an abstract contract, the method doesn't return `this` it returns a new instance so the AbstractCursor would have:
```ts
abstract clone(): AbstractCursor;
```
and FindCursor etc. can have the more specific subclass return type.

Reply:
duh, I should have thought of that :) implemented